dotCMS/src/main/java/com/dotcms/security/apps/AppsAPIImpl.java

```diff
@@ -565,35 +537,73 @@ private static String getServiceDescriptorDirectory() {
     }
 
    private boolean validateServiceDescriptor(final AppDescriptor appDescriptor)
-           throws DotDataException {
+           throws DotDataValidationException {
        if(UtilMethods.isNotSet(appDescriptor.getKey())){
-          throw new DotDataException("The required field `key` isn't set on the incoming file.");
+          throw new DotDataValidationException("The required field `key` isn't set on the incoming file.");
        }
 
        if(appDescriptor.getKey().length() > 100){
-           throw new DotDataException("The required field `key` is too large.");
+           throw new DotDataValidationException("The required field `key` exceeds 100 chars length.");
        }
 
        if(UtilMethods.isNotSet(appDescriptor.getName())){
-           throw new DotDataException("The required field `name` isn't set on the incoming file.");
+           throw new DotDataValidationException("The required field `name` isn't set on the incoming file.");
        }
 
        if(UtilMethods.isNotSet(appDescriptor.getDescription())){
-           throw new DotDataException("The required field `description` isn't set on the incoming file.");
+           throw new DotDataValidationException("The required field `description` isn't set on the incoming file.");
        }
 
        if(UtilMethods.isNotSet(appDescriptor.getIconUrl())){
-           throw new DotDataException("The required field `iconUrl` isn't set on the incoming file.");
+           throw new DotDataValidationException("The required field `iconUrl` isn't set on the incoming file.");
        }
 
        if(!UtilMethods.isSet(appDescriptor.getParams())){
-           throw new DotDataException("The required field `params` isn't set on the incoming file.");
+           throw new DotDataValidationException("The required field `params` isn't set on the incoming file.");
+       }
+
+       for (Map.Entry<String, ParamDescriptor> entry : appDescriptor.getParams().entrySet()) {
+           validateParamDescriptor(entry.getKey(), entry.getValue());
        }
 
        return true;
 
    }
 
+   private void validateParamDescriptor(final String name, final ParamDescriptor descriptor) throws DotDataValidationException {
+
+       if(UtilMethods.isNotSet(name)){
+           throw new DotDataValidationException("Param descriptor is missing required  field `name` .");
+       }
+
+       if(name.length() > 100){
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid using Literals in Conditional Statements](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5358201)